hoomd/hpmc/UpdaterShape.h

```diff
@@ -0,0 +1,501 @@
+// Copyright (c) 2009-2022 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#ifndef _UPDATER_SHAPE_H
+#define _UPDATER_SHAPE_H
+
+#include "IntegratorHPMCMono.h"
+#include "ShapeMoves.h"
+#include "ShapeUtils.h"
+#include "hoomd/HOOMDMPI.h"
+#include "hoomd/Updater.h"
+#include <algorithm>
+#include <numeric>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace hoomd
+    {
+
+namespace hpmc
+    {
+
+template<typename Shape> class UpdaterShape : public Updater
+    {
+    public:
+    UpdaterShape(std::shared_ptr<SystemDefinition> sysdef,
+                 std::shared_ptr<Trigger> trigger,
+                 std::shared_ptr<IntegratorHPMCMono<Shape>> mc,
+                 std::shared_ptr<ShapeMoveBase<Shape>> move);
+
+    ~UpdaterShape();
+
+    virtual void update(uint64_t timestep);
+
+    void initializeDeterminatsInertiaTensor();
+
+    Scalar getTotalParticleVolume()
+        {
+        Scalar volume = 0.0;
+        ArrayHandle<unsigned int> h_ntypes(m_ntypes, access_location::host, access_mode::read);
+        for (unsigned int ndx = 0; ndx < m_ntypes.getNumElements(); ndx++)
+            {
+            detail::MassProperties<Shape> mp(m_mc->getParams()[ndx]);
+            volume += mp.getVolume() * Scalar(h_ntypes.data[ndx]);
+            }
+        return volume;
+        }
```

Comment:
I think a per-type single particle volume is more useful.

Reply:
see https://github.com/glotzerlab/hoomd-blue/pull/989#discussion_r846659044